### PR TITLE
Update GraalVM to 20.0.0

### LINKED ISSUE #520: Bump GraalVM Version to 20.0
### Summary
GraalVM 20.0 has just been released, and we should upgrade at some point to the new version. Release notes for the version can be found [here](https://www.graalvm.org/docs/release-notes/20_0/#2000), and it doesn't seem that there are any breaking API changes. However, the deprecation of the `-Dgraal` command-line options is going to require some work.

### Value
We keep up with the GraalVM release cycle, thereby reducing the effort to upgrade when a release comes out with a feature that we could like to use. 

### Specification

- [ ] Bump the GraalVM version and API versions in build.sbt and the CI configuration.
- [ ] Ensure that everything still builds and use the benchmark suite to characterise any performance regressions.
- [ ] Move the `-Dgraal` options we use into the polyglot API bindings, ensuring that no additional regressions occur.

### Acceptance Criteria & Test Cases

- Enso builds and runs using GraalVM 20.0.

### PR DESCRIPTION
**Pull request by rzats**
### Pull Request Description
Closes #520.

### Important Notes

The benchmark results for both v19 and v20 are very similar on my machine, besides a few highlighted exceptions. See the original (and very verbose) output logs [here.](https://gist.github.com/rzats/a40dd575ded344928f96eb5ad5d57526)

| Benchmark  | GraalVM 19.3.1 | GraalVM 20.0.0 |
| ------------- | ------------- | ------------- |
| AtomBenchmarks.benchGenerateList | 15.458242904915702  | 16.413785484116907 | 
| AtomBenchmarks.benchMapReverseCurryList | 22.969261846144597  | 21.243586123728388 |
| AtomBenchmarks.benchMapReverseList | 20.57017271427409  | 21.989881841025646  |
| ***AtomBenchmarks.benchReverseList*** | ***35.51495561110558***  | ***19.47601254844943***  |
| AtomBenchmarks.benchReverseListMethods | 20.97737287136907  | 19.92246298164083  |
| ***AtomBenchmarks.benchSumList*** | ***17.263403325563722***  | ***6.457845852615831***  |
| AtomBenchmarks.benchSumListFallback | 7.236245815898873  | 6.131506735199339  |
| AtomBenchmarks.benchSumListMethods | 8.454580947887866  | 8.151325625560538 |
| AtomBenchmarks.sumListLeftFold | 6.523510021335296  | 6.670444832771824  |
| NamedDefaultedArgumentBenchmarks.benchSumTCOWithDefaultArgs | 51.03333002135889  | 50.87828658811418  |
| ***NamedDefaultedArgumentBenchmarks.benchSumTCOWithNamedArgs*** | ***24.357791573350955***  | ***50.55821447211682***  |
| RecursionBenchmarks.benchNestedThunkSum | 36.38848489720949  | 41.17952319258729  |
| RecursionBenchmarks.benchOversaturatedRecursiveCall | 48.4766120820271  | 49.71516899021767  |
| RecursionBenchmarks.benchSumRecursive | 7.174044739876558E-4  | 7.320496063816858E-4  |
| RecursionBenchmarks.benchSumStateTCO | 48.536994919666064  | 48.57687544242765  |
| RecursionBenchmarks.benchSumTCO | 48.63046886711716  | 48.68137251673151  |
| RecursionBenchmarks.benchSumTCOFoldLike | 48.66120624001649  | 48.590116440556265  |
| RecursionBenchmarks.benchSumTCOWithEval | 50.039484320893564  | 48.70605361556713  |

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md), [Rust](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

File: engine/runtime/src/test/scala/org/enso/interpreter/test/InterpreterTest.scala [hunk 1]
```diff
@@ -47,7 +47,7 @@ trait InterpreterRunner {
       )
   }
   val output           = new ByteArrayOutputStream()
-  val ctx              = Context.newBuilder(LanguageInfo.ID).out(output).build()
+  val ctx              = Context.newBuilder(LanguageInfo.ID).allowExperimentalOptions(true).out(output).build()
```
Comment (rzats) [inline engine/runtime/src/test/scala/org/enso/interpreter/test/InterpreterTest.scala]: (`Dpolyglot.engine.IterativePartialEscape` is considered experimental)

File: azure-pipelines.yml [hunk 2]
```diff
@@ -142,6 +142,7 @@ jobs:
           echo ##vso[task.setvariable variable=JAVA_HOME]C:\$(graalDistributionName)-$(graalVersion)
           echo ##vso[task.setvariable variable=PATH]C:\$(graalDistributionName)-$(graalVersion)\bin;%PATH%
           echo ##vso[task.setvariable variable=PATH]%USERPROFILE%\sbt\bin\;%PATH%
+          echo ##vso[task.setvariable variable=TRUFFLE_STRICT_OPTION_DEPRECATION]true
```
Comment (rzats) [inline azure-pipelines.yml]: This is to make sure that the deprecated flags are actually deprecated everywhere. Could be removed after a CI run?

Comment (kustosz) [reply azure-pipelines.yml]: Let's keep it here for now, in case we introduce a deprecated option by accident in the future.

**Comment by kustosz** [comment]
Great contribution, thank you very much! :)
Merging as soon as the CI passes.

**Comment by iamrecursion** [comment]
And it's all on master! Thank you so much for getting this sorted for us.

**Pull request merged**